introspection-engine/connectors/sql-introspection-connector/src/re_introspection.rs

```diff
@@ -127,10 +129,28 @@ fn keep_index_ordering(old_data_model: &Datamodel, new_data_model: &mut Datamode
 fn merge_relation_fields(old_data_model: &Datamodel, new_data_model: &mut Datamodel, warnings: &mut Vec<Warning>) {
     let mut changed_models = BTreeSet::new();
 
+    // Maps a model name to the table name it was introspected to. This is helpful when @@map is used.
+    // E.g., for
+    //
+    // ```prisma
+    // model Foo {
+    //     id     Int @id
+    //     bar    Bar @relation(fields: [bar_id], references: [id])
+    //     bar_id Int @unique
+    //     @@map("foo_table")
+    // }
+    //
+    // the map would be {"Foo" -> "foo_table"}.
+    // ```
+    let old_model_name_to_final_database_name: HashMap<String, String> = old_data_model
+        .models()
+        .map(|m| (m.name.clone(), String::from(m.final_database_name())))
+        .collect();
```

Comment:
Any suggestion on how to avoid these clones while still being able to use the function in the following closures is well accepted.

Reply:
I wouldn't worry about it here, but you can change line 147 to `        .map(|m| (m.name.as_str(), m.final_database_name()))` and the type ascription to `HashMap<&str, &str>`. To optimize further, you could make the HashMap potentially a lot smaller if you `.filter()` for only the models with a mapped name first — then a model missing in the map would mean the model is not remapped, which would be handled with the existing code lower down.

But we shouldn't worry too much here, we're in the process of rewriting this file and the string comparisons will go away, we'll have a big map of integer identifiers for these lookups, since they happen repeatedly during introspection and reintrospection.

Reply:
Thanks. I have simply got rid of `.clone()` in favor of `.as_str()`. I will skip other optimizations for now, as we expect this file to change significantly already